syntax = "proto3";

package media;

// I don't care that media is already a latin plural, Oxford dictionnary has accepted the new "medias" plural form 
service Medias {
  // Sends a greeting
  rpc GetAllMedias (GetAllMediasRequest) returns (stream MediaResponse) {}
  rpc GetMediaById (GetMediaByIdRequest) returns (MediaResponse) {}
  rpc AddMedia (stream AddMediaRequest) returns (stream AddMediaResponse) {}
  rpc DeleteMedia (DeleteMediaRequest) returns (DeleteMediaResponse) {}
}

message Media {
	int32 id = 1;
	string file_uri = 2;
	int32 file_type = 3;
	string thumbnail_uri = 4;
}
message GetAllMediasRequest {}

message MediaResponse {
	bool success = 1;
	Media media = 2;
}
message GetMediaByIdRequest {
	int32 id = 1;
}
message AddMediaRequest {
	Media media = 1;
}
message AddMediaResponse {
	string message = 1;
	int32 count = 2;
}
message DeleteMediaRequest {
	int32 id = 1;
}
message DeleteMediaResponse {
	Media media = 1;
}
